packages/core/strapi/lib/middlewares/session.js

```diff
@@ -0,0 +1,45 @@
+'use strict';
+
+const crypto = require('crypto');
+const { defaultsDeep, isEmpty, omit, has } = require('lodash/fp');
+const session = require('koa-session');
+
+const defaultConfig = {
+  key: 'koa.sess',
+  maxAge: 86400000,
+  autoCommit: true,
+  overwrite: true,
+  httpOnly: true,
+  signed: true,
+  rolling: false,
+  renew: false,
+  secure: false,
+  sameSite: null,
+};
+
+module.exports = (userConfig, { strapi }) => {
+  if (isEmpty(strapi.server.app.keys)) {
+    let secretKeys = [];
+
+    if (has('secretKeys', userConfig)) {
+      secretKeys = userConfig.secretKeys;
+    } else if (has('SESSION_SECRET_KEYS', process.env)) {
+      secretKeys = process.env.SESSION_SECRET_KEYS.split(',');
+    } else {
+      // auto generate secret keys if they are not provided
+      for (let i = 0; i < 4; i++) {
+        secretKeys.push(crypto.randomBytes(64).toString('hex'));
+      }
+      strapi.fs.appendFile('.env', `SESSION_SECRET_KEYS=${secretKeys.join(',')}\n`);
```

Comment:
As mentioned in DMs, we will need to handle some read-only mode for the .env file eventually. let's keep this in mind

Reply:
Yes, we are doing that in a few places (U&P, api token, session middlewares). No issues from the community so far?

Reply:
There is an issue with how we are writing to the .env yes. It was just mentioned in Discord and I'm having them create a bug report.

![image](https://user-images.githubusercontent.com/8593673/146830476-ba93efa9-5654-4a8c-83bf-2358408a69d5.png)

I also don't really care for how we are writing these either and we do it for some but not others, it's quite odd.

Reply:
Interesting, @derrickmehaffy do you have any recommendation on how to handle this usecase?
We need the user to provide some secrets in order for Strapi to properly work. It is easier if they are autogenerated, but where should we store them and with which logic?

Reply:
TBH I don't like that we are auto-writing to a .env, this is not something you usually do without asking the user.

I think every single one of these needs to have a config file option (server, admin, api, plugins, middlewares, whatever) by default and we set the generic random as the 2nd param like we used to. If they aren't using quickstart then we can ask at the CLI if they want a .env file but not every reboot.

Reply:
Certain providers (like google) have very strict read-only filesystems that auto-writing files simply will not work and just crash the application.

Reply:
FYI another user just had an issue on this. I'm having them open a github issue on it
![image](https://user-images.githubusercontent.com/8593673/147962693-2c561fc9-011c-492d-a5c3-1392f0ea26c8.png)

![image](https://user-images.githubusercontent.com/8593673/147962703-7e67778f-8aca-4768-adcb-cfe62821da27.png)

They are declaring the var using env vars and don't need the .env but we are trying to write it anyway on a system that is a enforced ROFS

Reply:
We only write the key if it's not defined in the conf or in the end vars. Which means it should be good for production as you should only deploy once you have set your env vars anyway. 

We will handle this in another PR so we can move forward with this one